web/packages/teleport/src/Discover/useDiscover.tsx

```diff
@@ -231,6 +247,14 @@ export function DiscoverProvider({
     });
   }
 
+  // trigger update Discover flow.
+  useEffect(() => {
+    if (updateFlow && updateFlow.agentMeta) {
+      updateAgentMeta(updateFlow.agentMeta);
+      onSelectResource(updateFlow.resourceSpec);
+    }
+  }, [updateFlow]);
```

Comment:
This is a react anti pattern https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes.
Maybe we could replace it with setting initial value in `setState`s?

Reply:
I am not sure that suggestion fits the bill here.
`onSelectResource` is not a react `SetStateAction` but  a function that starts discover flow. It, among setting up resource view, also dispatches Event, which is an outgoing network request.  Only `updateAgentMeta` directly wraps a SetStateAction.